magenta-lib/src/main/scala/magenta/input/resolver/DeploymentResolver.scala

```diff
@@ -1,38 +1,41 @@
 import magenta.input.{ConfigError, Deployment, DeploymentOrTemplate, RiffRaffDeployConfig}
 
 object DeploymentResolver {
-  def resolve(config: RiffRaffDeployConfig): List[Either[ConfigError, Deployment]] = {
-    config.deployments.map { case (label, rawDeployment) =>
+  def resolve(config: RiffRaffDeployConfig): ValidatedNel[ConfigError, List[Deployment]] = {
+    config.deployments.traverseU[ValidatedNel[ConfigError, Deployment]] { case (label, rawDeployment) =>
       for {
-        templated <- applyTemplates(label, rawDeployment, config.templates).right
-        deployment <- resolveDeployment(label, templated, config.stacks, config.regions).right
-        _ <- validateDependencies(label, deployment, config.deployments).right
-      } yield deployment
+        templated <- applyTemplates(label, rawDeployment, config.templates)
+        deployment <- resolveDeployment(label, templated, config.stacks, config.regions)
+        validatedDeployment <- validateDependencies(label, deployment, config.deployments)
+      } yield validatedDeployment
     }
   }
 
   /**
     * Validates and resolves a templated deployment by merging its
     * deployment attributes with any globally defined properties.
     */
-  private[input] def resolveDeployment(label: String, templated: DeploymentOrTemplate, globalStacks: Option[List[String]], globalRegions: Option[List[String]]): Either[ConfigError, Deployment] = {
-    for {
-      deploymentType <- templated.`type`.toRight(ConfigError(label, "No type field provided")).right
-      stacks <- templated.stacks.orElse(globalStacks).toRight(ConfigError(label, "No stacks provided")).right
-      regions <- templated.regions.orElse(globalRegions).toRight(ConfigError(label, "No regions provided")).right
-    } yield {
+  private[input] def resolveDeployment(label: String, templated: DeploymentOrTemplate, globalStacks: Option[List[String]], globalRegions: Option[List[String]]): ValidatedNel[ConfigError, Deployment] = {
+    (Validated.fromOption(templated.`type`, NEL.of(ConfigError(label, "No type field provided"))) |@|
+      Validated.fromOption(templated.stacks.orElse(globalStacks), NEL.of(ConfigError(label, "No stacks provided"))) |@|
+      Validated.fromOption(templated.regions.orElse(globalRegions), NEL.of(ConfigError(label, "No regions provided")))) map { (deploymentType, stacks, regions) =>
       Deployment(
-        name              = label,
-        `type`            = deploymentType,
-        stacks            = stacks,
-        regions           = regions,
-        actions           = templated.actions,
-        app               = templated.app.getOrElse(label),
-        contentDirectory  = templated.contentDirectory.getOrElse(label),
-        dependencies      = templated.dependencies.getOrElse(Nil),
-        parameters        = templated.parameters.getOrElse(Map.empty)
+        name = label,
+        `type` = deploymentType,
+        stacks = NEL.fromListUnsafe(stacks),
```

Comment:
Couldn't this `Unsafe` be avoided by lifting into the NEL within the `Validated` clause? 

e.g.

```scala
Validated.fromOption(templated.stacks.orElse(globalStacks).flatMap(NEL.fromList), NEL.of(ConfigError(label, "No stacks provided")))
```